[IMP] sign: 19.2 updates (#17436)
odoo/documentation: content/applications/productivity/sign/request_signatures.rst

.. note::
   When you open a template that has already been used to send a signature request, the template is

>    in *visualization mode*. Click :icon:`fa-pencil` :guilabel:`Duplicate & Edit` to create a copy

Reviewer: I missed this before, and only spotted it when I looked more closely at the image again. I guess it would be more correct now to write:

```suggestion
   in :guilabel:`View only mode`. Click :icon:`fa-pencil` :guilabel:`Duplicate & Edit` to create a copy
```

- :ref:`Prepare the document(s) for signing <sign/request-signatures/prepare-document>` by adding
  or editing fields, and, if relevant, adding or removing signers and/or documents.
- Optionally, :ref:`define or edit settings for specific signers

>   <sign/request-signatures/signer-settings>`, such as requiring additional authentication, allowing
>   a signer to :ref:`delegate their signature <sign/sign-document/delegate>` and to be :ref:`replaced
>   <sign/request-signatures/change-signers>`, or assigning a fixed signer.

Reviewer: ```suggestion
  <sign/request-signatures/signer-settings>`, such as requiring additional authentication, allowing
  a signer to :ref:`delegate their signature <sign/sign-document/delegate>`, allowing a signer to be :ref:`replaced
  <sign/request-signatures/change-signers>`, or assigning a fixed signer.
```

If I understand correctly, allowing a signer to delegate their own signature and allowing anyone with the appropriate rights to replace a signer are two different things, right? And both are only possible if 'Can delegate' has been enabled.

Author: Yes exactly, hence why I originally chose to link these two items with an *and* to try and imply that while those are two different things, they are linked (you can't allow to delegate without allowing to replace, and vice versa).

Reviewer: The first time I read it, I felt it _could_ have been interpreted as '... allowing a signer to delegate their signature and [therefore] to be replaced', rather than these being things that are initiated by different people.

Author: Ohhh! Ah I understand better now ... what about using and/or? it's not the prettiest but do you think that'd help?

- Optionally, define or edit general settings for the template by clicking the :icon:`fa-cog`
  :guilabel:`(cog)` icon beside the template name, then clicking :icon:`fa-cog`
  :menuselection:`Configuration`:

  - By default, the :guilabel:`Name` of the template is the name of the file that was uploaded,
    or, in the case of a :ref:`document envelope <sign/request-signatures/create-envelope>`, the
    name of the first file uploaded. If desired, edit the name of the template.
  - :guilabel:`Available in`: Select the :doc:`Odoo model <../../studio/models_modules_apps>` to
     which the template is linked to restrict the template to :ref:`signatures requests made from
     records of that model <sign/request-signatures/template-odoo-record>`. Leave the field empty to

>      make the template available from any model.

Reviewer: ```suggestion
  - :guilabel:`Available in`: Select the :doc:`Odoo model <../../studio/models_modules_apps>` to
    which the template is linked to restrict the template to :ref:`signatures requests made from
    records of that model <sign/request-signatures/template-odoo-record>`. Leave the field empty to
    make the template available from any model.
```

Just some extra spaces messing with the formatting.

Author: Good catch, thanks!

  - :guilabel:`Documents folder`: Define in which :ref:`folder in the Documents app
    <documents/folders>` the documents using this template should be saved once fully signed.
  - Click :guilabel:`More options` to access and edit :ref:`advanced template settings
    <sign/request-signatures/template-form>`.

.. tip::
   To replace a document in the template, such as with an updated version, while preserving the
   fields already added, click the :icon:`fa-ellipsis-v` (:guilabel:`vertical ellipsis`) icon next
   to the relevant document, select :guilabel:`Replace document`, choose the new document, then
   click :guilabel:`Replace document`. Each replacement creates a duplicate of the template with
   the fields placed at the same position as in the original. If the new document has a different

>    layout or added text, the fields may no longer align correctly. Reposition them as needed.

Reviewer: ```suggestion
   To replace a document in the template, such as with an updated version, while preserving the
   fields already added, click the :icon:`fa-ellipsis-v` (:guilabel:`vertical ellipsis`) icon next
   to the relevant document, select :guilabel:`Replace document`, choose the new document, then
   click :guilabel:`Replace document`. 
   
   Each replacement creates a duplicate of the template with
   the fields placed at the same position as in the original. If the new document has a different
   layout or added text, the fields may need to be repositioned.
```

1) I think this block is a bit too big to keep in one piece, so I'd suggested starting a new paragraph at 'Each replacement ...'
2) About repositioning: maybe worth (adding an anchor to and) linking to this content (in the 'Add fields to a document' part), and to think about moving the tip so it's closer to the point about repositioning? I'm thinking that if someone needs to be moving around fields on such a replacement document, they might appreciate the help

<img width="1028" height="735" alt="Image" src="https://github.com/user-attachments/assets/42e60a2a-6774-434d-ab59-6ffa53375a7a" />

   - Remove the :guilabel:`My Favorites` :ref:`filter <search/preconfigured-filters>` to view all
     document templates in the database.
   - A panel on the left side allows the view to be filtered by :ref:`tag
     <sign/configuration/tags>` if the displayed templates have been assigned tags. Click the
     :icon:`oi-panel-right` icon to close or open the panel.

For each template, the following information is shown:

- the name of the template
- the month and year in which the template was created
- the avatar of the template's :guilabel:`Owner`; by default, the owner is the user who created the

>   template. This can be changed in the :ref:`configuration settings of the template

Reviewer: Just 'template settings' rather than 'configuration settings of the template'?

Hovering over the template's card reveals a :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)`
icon in the top-right corner. Click the icon to:

> - :ref:`Edit a template <sign/request-signatures/edit-template>` that has *not yet been used* to
>   send a signature request. Alternatively, click on a template to open it for editing.

Reviewer: Maybe after this point we should add a point about the 'Stop Sharing' option, which I guess only appears if there is a template for which a sharing link has been created. 

Something like:

```suggestion
- :guilabel:`Stop Sharing` the link to the templated document or document envelope, if relevant; after sharing is stopped, the link can no longer be opened.
```

- :guilabel:`Duplicate` a template. The duplicated template can then be :ref:`edited
  <sign/request-signatures/edit-template>` as needed.
- :guilabel:`Archive` a template that is not currently needed.

> - Modify a template's :ref:`configuration settings <sign/request-signatures/template-form>`.

Reviewer: As before, just 'settings' rather than 'configuration settings'?

- Select a :guilabel:`Color` to add a colored bar to the left side of a card to visually
  differentiate it.

> Export template answers

Reviewer: General comment: I know they've used the term 'answers' in the app, but I really don't think that works well. Could we use 'response(s)' except when referring to actual GUI labels that use 'answers'?

For this title, I'm not sure we need to mention 'template', since we're in the 'Manage templates' section, and the first sentence explains/will explain it clearly enough.

So, my suggestions for a title are:

- Export responses
- Export response data
- Export signer responses

~~~~~~~~~~~~~~~~~~~~~~~

> Answers from all signature requests sent using a specific template can be exported, provided the

Reviewer: I'm not sure that '... from all signature requests sent' really captures this accurately enough, since it's more like 'responses entered by signers in template-based documents or document envelopes sent via a signature request...'. 

Maybe it could be something as simple as:

- All signer responses related to a specific template can be exported, provided the Documents app is installed.

Lastly, for this one ... I'm wondering if it might be useful here to give an example of what this could be used for?

:doc:`../documents` app is installed. To export them, go to :menuselection:`Sign --> Templates`,
click the relevant template's card, then click the :icon:`fa-cog` (:guilabel:`Actions`) icon and
select :guilabel:`Export Answers`.

> Answers are exported to a :doc:`../spreadsheet`, with one column per :ref:`field

Reviewer: ```suggestion
Answers are exported to an :doc:`Odoo spreadsheet <../spreadsheet>`, with one column per :ref:`field
```

<sign/request-signatures/add-fields>`, one line per :ref:`signer <sign/request-signatures/signers>`,
and one tab per document for :ref:`document envelopes <sign/request-signatures/create-envelope>`.

.. note::
   The spreadsheet is named after the template and saved in the :guilabel:`Sign Request Answers`

>    folder in Documents.

Reviewer: Maybe it's just me, but I feel 'is named after' is more for people, e.g., naming a child after a grandparent. But ... I'm struggling to find a concise way to word this. Something like this maybe?

```suggestion
.. note::
   The spreadsheet is saved in the :guilabel:`Sign Request Answers`
   folder in Documents, with the template name referenced in the name of the spreadsheet.
```

.. _sign/request-signatures/template-form:

> Template configuration settings

Reviewer: Just 'Template settings'?

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To access and modify a template's settings, go to :menuselection:`Sign --> Templates`, hover over
the relevant template's card, click the :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)` icon,
and select :guilabel:`Configuration`.

The following fields are available on the form:

- the template's name
- :guilabel:`Available in`: Select the :doc:`Odoo model <../../studio/models_modules_apps>` to
  which the template is linked to restrict the template to :ref:`signature requests made from
  records of that model <sign/request-signatures/template-odoo-record>`. Leave the field empty to
  make the template available from any model.
- :guilabel:`Authorized Users`: Specify the users authorized to view and use the template.
- :guilabel:`Authorized Groups`: If :guilabel:`Manage template access` is enabled in the
  :ref:`main Sign settings <sign/configuration/settings>`, define the user groups whose members are
  authorized to view and use this template.
- :guilabel:`Owner`: The user authorized to modify the template. By default, this is the user

>   who created it. Select a different user to transfer ownership.

Reviewer: I don't think it's only the Owner who can modify the template. In my runbot, I set Marc Demo as a second Authorized User and it seems he can do anything (add and remove fields and signers, update the template settings) except link the template to a model. Maybe you had a different experience? Or maybe it _should_ only be the Owner who can modify the template?

Author: You're totally right (and the way this works really makes no sense to me but 🤐 )
Apparently it will be changed in 19.4 ... Anyways, I changed the desription of the Owner and Authorized users fields and will keep it very simple until they make this (slightly) more logical, if that's OK with you?

- :guilabel:`CC`: Add contacts or email addresses to notify when the document is fully signed or
  :ref:`refused <sign/sign-document/decline>`.
- :guilabel:`Valid for`: Set a default number of days after which signature requests using this
  template expire; enter `0` if signature requests should not expire.

  .. note::
     The expiry date and CC contacts of a signature request can always be modified when
     :ref:`configuring the signature request <sign/request-signatures/configure-request>`.

- :guilabel:`Tags`: Add :ref:`tags <sign/configuration/tags>` to categorize the
  template.
- :guilabel:`Redirect Link`: Add a URL to redirect signers after signing the document, such as a
  *Thank you* page, or the website's homepage.
- In the :guilabel:`Communication` tab, type a default text to include when a signature request
  using this template is sent via email. The text can still be edited before sending.
- In the :guilabel:`Signers` tab, view and edit any :ref:`signer settings
  <sign/request-signatures/signer-settings>` defined for the template.

.. tip::
   - Access all signature requests linked to the template by clicking the :guilabel:`Signed

>      Document` smart button.

Reviewer: I guess that, like before, these aren't actually the _signature requests_ that appear, but rather the fully signed documents (e.g., if you share a link to a templated document with 50 people and 25 sign the document, those 25 'documents' will show when you click 'Signed Document'; or if you individually send signature requests using that template to 5 people, it's only the signed documents - and not the signature requests - that appear)

Hovering over the template's card reveals a :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)`
icon in the top-right corner. Click the icon to:

- :ref:`Edit a template <sign/request-signatures/edit-template>` that has *not yet been used* to
  send a signature request. Alternatively, click on a template to open it for editing.
- :guilabel:`Stop Sharing` the link to the templated document or document envelope, if relevant;
  after sharing is stopped, the link can no longer be opened.
- :guilabel:`Duplicate` a template. The duplicated template can then be :ref:`edited
  <sign/request-signatures/edit-template>` as needed.
- :guilabel:`Archive` a template that is not currently needed.

> - Modify a template's :ref:`settings <sign/request-signatures/template-form>`.

Reviewer: Sorry, my suggestion here overlooked the fact that you're actually clicking on **Configuration** to modify the template's settings. So maybe:

```suggestion
- Modify a template's :ref:`settings <sign/request-signatures/template-form>` by clicking :guilabel:`Configuration`.
```

Or there's maybe a better way ...

Responses are exported to an :doc:`Odoo spreadsheet <../spreadsheet>`, with one column per
:ref:`field <sign/request-signatures/add-fields>`, one line per
:ref:`signer <sign/request-signatures/signers>`, and one tab per document for :ref:`document
envelopes <sign/request-signatures/create-envelope>`.

.. note::
   The spreadsheet is saved in the :guilabel:`Sign Request Answers` folder in Documents, with the
   template name referenced in the name of the spreadsheet.

.. _sign/request-signatures/template-form:

Template settings
~~~~~~~~~~~~~~~~~

To access and modify a template's settings, go to :menuselection:`Sign --> Templates`, hover over
the relevant template's card, click the :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)` icon,
and select :guilabel:`Configuration`.

The following fields are available on the form:

- the template's name
- :guilabel:`Available in`: Select the :doc:`Odoo model <../../studio/models_modules_apps>` to
  which the template is linked to restrict the template to :ref:`signature requests made from
  records of that model <sign/request-signatures/template-odoo-record>`. Leave the field empty to
  make the template available from any model.

> - :guilabel:`Authorized Users`: Specify the users authorized to modify the template.

Reviewer: Sorry, but I'm not remembering now why you were only mentioning 'modify' here, and not something like 'view, use, and modify' ...? (because you need to be an authorized user to even see the template, right?)

Author: Whoops, I went to fast, sorry!

To define additional configuration for signers, click the :icon:`fa-ellipsis-v` :guilabel:`(vertical
ellipsis)` icon on the row of the signer. For each signer, the following options are available:

- :guilabel:`Extra Authentication Step`: Require the signer to :ref:`authenticate using a specific
  authentication method <sign/security/authentication>`.

> - :guilabel:`Can delegate`: Allow the signer to :ref:`delegate their signature
>   <sign/sign-document/delegate>` and :ref:`be replaced after the signature request has been sent
>   <sign/request-signatures/change-signers>`.

Reviewer: ```suggestion
- :guilabel:`Can delegate`: Allow the signer to :ref:`delegate their signature
  <sign/sign-document/delegate>` or :ref:`be replaced after the signature request has been sent
  <sign/request-signatures/change-signers>`.
```

I think the 'and' should be 'or', no? Becuase in my mind we're talking about two different things here: me delegating my signature and thereby nominating someone else to replace me ... OR ... the sender of a signature request replacing a signer, e.g., because they were informed that someone left the company. To me, with the 'and' it reads like the two parts of the sentence are part of the same thing.

Author: I get your point, but I guess this overlaps with [this](https://github.com/odoo/documentation/pull/17436/changes#r3187319884)
Technically, you could also do both (the signer can delegate and then I can replace the person they delegated to; it probably doesn't make a lot of sense flow-wise, but it is technically possible). The two features are tied together. Maybe we should discuss this?

- :ref:`set a Valid Until date <sign/request-signatures/validity>` after which the signature request
  expires
- :ref:`set Reminders <sign/request-signatures/validity>` to be sent at fixed intervals
- enable :guilabel:`Hide certificate key pages` to hide the :ref:`reference of the certificate of

>   completion <sign/security/hash>` displayed by default on each page of the document(s)

Reviewer: ```suggestion
- enable :guilabel:`Hide certificate key on pages` to hide the :ref:`reference of the certificate of
```

Also, I think the ref is to slightly the wrong place. It should be to 'sign/security/certificate-of-completion'.

The signatory hash is related to each individual signer, if I understand right, while the certificate reference is really at the level of the fully signed document.  

In the screenshots, the red box is the (same) certificate reference that can be hidden on the signed doc and that appears on the certificate of completion, while the green boxes are the signatory hashes that only appear on the certificate of completion.

_Screenshot 1_
<img width="776" height="389" alt="Image" src="https://github.com/user-attachments/assets/c30a0d63-3a32-4cc3-ae95-259b2f9ac366" />

_Screenshot 2_
<img width="1110" height="691" alt="Image" src="https://github.com/user-attachments/assets/37c3951f-032b-41e2-b9a0-43d507dd004e" />

  - When a signer has fully signed the document(s), the checkbox is ticked, the signer's name
    appears in green, and the date of the signature is added after the name.

>   - When a signature request is canceled, the signers name appears in red with a :icon:`fa-times`

Reviewer: ```suggestion
  - When a signature request is canceled, the signer's name appears in red with a :icon:`fa-times`
```

Author: And while checking this, I realized that if one signer has already signed the signature request before it was canceled, their name is still in green with a check mark; only signers who haven't signed yet appear in red but that kinda makes sense to me so I'm not sure I should describe this more?
EDIT: I changed this
